core-bundle/src/Controller/ContentElement/DownloadsController.php

```diff
@@ -222,16 +222,19 @@ private function getPreviews(FilesystemItem $filesystemItem, ContentModel $model
         }
     }
 
-    private function handleDownload(Request $request): void
+    private function handleDownload(Request $request, ContentModel $model): void
     {
         $response = $this->fileDownloadHelper->handle(
             $request,
             $this->filesStorage,
-            function (FilesystemItem $item, array $context): Response|null {
-                if (
-                    null === ($model = $this->getContaoAdapter(ContentModel::class)->findById($context['id'] ?? null)) ||
-                    !$this->getFilesystemItems($model)->any(static fn (FilesystemItem $listItem) => $listItem->getPath() === $item->getPath())
-                ) {
+            function (FilesystemItem $item, array $context) use ($model): Response|null {
+                // Do not handle downloads from other DownloadController
+                // elements on the same page (see #5568)
+                if ($model->id !== ($context['id'] ?? null)) {
+                    return new Response('', Response::HTTP_NO_CONTENT);
```

Comment:
@m-vo shouldn't that rather end in a 404? Sorry for being late to the party 🙈

Reply:
AFAIK the empty response is ignored later in the code and never sent to the client.

Reply:
Correct. And additionally, 404 would be wrong, because this is not a "file not found" case but a "not responsible for this file" case.

Reply:
Yes, we are only going to return those responses once switching to a real controller. (If we manage to solve permissions…)